NuRadioReco/modules/RNO_G/stationHitFilter.py

```diff
@@ -0,0 +1,462 @@
+                    self._passed_time_checker = True
+                    if not self._complete_time_check:
+                        break
+            else:
+                break
+
+        return self._passed_time_checker
+
+
+    def _hit_checker(self):
+        """
+        Find a high hit (maximum > 6.5*RMS) in all 15 in-ice channels.
+
+        See if there's at least 1 high hit in all input channels,
+        if yes then the event passes the hit checker.
+        When multiple thresholds are used, the hit checker then becomes more lenient;
+        for example, if there are two hits above the second-highest threshold but below the highest threshold,
+        the event will pass the hit checker too (when is_multi_thresholds = True).
+
+        Returns
+        -------
+        self._passed_hit_checker: bool
+            Event passes the hit checker or not
+        """
+        self._passed_hit_checker = False
+        envelopes = self.get_envelope_trace()
+        hit_thresholds = self.get_hit_thresholds()
+        n_counts_above_threshold = np.zeros(self._n_thresholds)
+
+        for i_channel in range(self._n_channels_in_ice):
+            for i in range(self._n_thresholds):
+                if np.amax(envelopes[i_channel]) > hit_thresholds[i_channel][i]:
+                    self._is_over_hit_threshold[i_channel][i] = True
+                    n_counts_above_threshold[i] += 1
+                else:
+                    self._is_over_hit_threshold[i_channel][i] = False
+
+        for i in range(self._n_thresholds):
+            if n_counts_above_threshold[i] > i:
+                self._passed_hit_checker = True
+
+        return self._passed_hit_checker
+
+
+    def begin(self):
+        """(Unused)"""
+        pass
+
+
+    def apply_hit_filter(self, set_of_traces, set_of_times, RMS=np.array([])):
+        """
+        See if the input event will survive the Hit Filter or not.
+
+        After the setup, it first checks with the time checker,
+        if event passed the time checker then it passes the Hit Filter;
+        if event failed the time checker, then checks with the hit checker.
+
+        Parameters
+        ----------
+        set_of_traces: 2-D array of floats
+            A set of input trace arrays of all 24 channels
+        set_of_times: 2-D array of floats
+            A set of input times arrays of all 24 channels
+        RMS: 1-D array of floats (default: np.array([]))
+            A set of input RMS values of all 24 channels
+        """
+        self._passed_hit_filter = False
+
+        self._set_up(set_of_traces, set_of_times, RMS)
+        self._time_checker()
+
+        if not self._complete_hit_check:
+            if self.is_passed_time_checker():
+                self._passed_hit_filter = True
+                return self._passed_hit_filter
+            else:
+                self._hit_checker()
+                if self.is_passed_hit_checker():
+                    self._passed_hit_filter = True
```

Comment:
I am a bit confused. In l265 it says `if not self._complete_hit_check`. Why would you still run it when the time_checker fails?